mex/common/context.py

```diff
@@ -32,3 +32,36 @@ def pop(self, cls: type[_SingletonT]) -> _SingletonT:
     def reset(self) -> None:
         """Remove all singleton instances from the store."""
         self._instances_by_class.clear()
+
+
+class SingleSingletonStore(Generic[_SingletonT]):
+    """Thin wrapper for storing a single thread-local singleton.
+
+    Stores only a single instance. Requested class must either be
+    the same or a parent class of the stored class.
+    """
+
+    def __init__(self) -> None:
+        """Create a new settings singleton store."""
+        self._singleton: _SingletonT | None = None
+
+    def load(self, cls: type[_SingletonT]) -> _SingletonT:
+        """Retrieve the settings for the given class or create a new one."""
+        if self._singleton is None:
+            self._singleton = cls()
+            return self._singleton
+        if not issubclass(type(self._singleton), cls):
+            raise TypeError(
```

Comment:
```suggestion
            raise RuntimeError(
```

i'd use a runtime error here

Reply:
Adapted in 97f19ff9d1f2239b9f1000c7b7ab103fb8ca77dc